src/vuln_analysis/utils/standard_library_cache.py

```diff
@@ -0,0 +1,117 @@
+Loads an auto-generated JSON of standard libraries for supported ecosystems
+and exposes a simple lookup with lightweight hit/miss counters. Every 5
+lookups, a short summary is logged. Designed to minimize external API usage.
+"""
+
+from __future__ import annotations
+
+import json
+from pathlib import Path
+import threading
+# No typing imports needed; we use builtin generics
+
+from vuln_analysis.logging.loggers_factory import LoggingFactory
+from vuln_analysis.utils.dep_tree import Ecosystem
+
+
+logger = LoggingFactory.get_agent_logger(__name__)
+
+
+class StandardLibraryCache:
+    """In-memory cache for standard library package/module names per ecosystem.
+    Implements singleton pattern to ensure single instance across the application.
+    """
+    _instance = None
+    _lock = threading.Lock()
+
+    def __new__(cls):
+        if cls._instance is None:
+            with cls._lock:
+                if cls._instance is None:
+                    cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self, json_path: str | Path | None = None) -> None:
+        if not hasattr(self, '_initialized'):
+            # Initialize only once
+            base_path = Path(__file__).resolve().parents[1]
+            default_json = base_path / "data" / "standard_libs" / "standard_libraries.json"
+            self.json_path = Path(json_path) if json_path else default_json
+
+            self._ecosystem_to_set: dict[str, set[str]] = {}
+            self._hits: int = 0
+            self._misses: int = 0
+            self._lookups_since_last_log: int = 0
+            self._data_lock = threading.RLock()
+            self._initialized = True
+            self._load()
+
+    @classmethod
+    def get_instance(cls):
+        """Get the singleton instance of StandardLibraryCache"""
+        return cls()
+
+    def _load(self) -> None:
+        try:
+            data = json.loads(self.json_path.read_text(encoding="utf-8"))
+        except FileNotFoundError:
+            logger.warning("Standard libraries JSON not found at %s", self.json_path)
+            data = {}
+        except json.JSONDecodeError as exc:
+            logger.error("Failed to parse standard libraries JSON: %s", exc)
+            data = {}
+
+        for eco_key in (Ecosystem.GO.value, Ecosystem.PYTHON.value, Ecosystem.JAVASCRIPT.value, Ecosystem.JAVA.value):
+            items: list[str] = data.get(eco_key, [])
+            normalized = {self._normalize_name(name) for name in items if isinstance(name, str)}
+            self._ecosystem_to_set[eco_key] = normalized
+
+    @staticmethod
+    def _normalize_name(name: str) -> str:
+        # Normalize to lowercase; trim whitespace
+        return name.strip().lower()
+
+    def is_standard_library(self, package_name: str, ecosystem: Ecosystem | None) -> bool:
+        if not package_name:
+            return False
+        if ecosystem is None:
+            return False
+
+        with self._data_lock:
```

Comment:
@RedTanny Do you using a lock here, to ensure that the thread is getting the most up to date cache copy of the standard library packages?  as there is not chance of overriding existing entry by another thread ( so there will be data consistency/integrity problems),